contrib/core-contract-tests/tests/pox-4/pox_StackStxCommand.ts

```diff
@@ -0,0 +1,166 @@
+      // `u1`.
+      period: this.period,
+      // A string representing the function where this authorization is valid.
+      // Either `stack-stx`, `stack-extend`, `stack-increase` or `agg-commit`.
+      topic: Pox4SignatureTopic.StackStx,
+      // The PoX address that can be used with this signer key.
+      poxAddress: this.wallet.btcAddress,
+      // The unique auth-id for this authorization.
+      authId: this.authId,
+      // The maximum amount of uSTX that can be used (per tx) with this signer
+      // key.
+      maxAmount: maxAmount,
+    });
+
+    // The amount of uSTX to be locked in the reward cycle. For this test, we
+    // will use the maximum amount of uSTX that can be used (per tx) with this
+    // signer key.
+    const amountUstx = maxAmount;
+
+    // Act
+    const stackStx = real.network.callPublicFn(
+      "ST000000000000000000002AMW42H.pox-4",
+      "stack-stx",
+      [
+        // (amount-ustx uint)
+        Cl.uint(amountUstx),
+        // (pox-addr (tuple (version (buff 1)) (hashbytes (buff 32))))
+        poxAddressToTuple(this.wallet.btcAddress),
+        // (start-burn-ht uint)
+        Cl.uint(real.network.blockHeight),
+        // (lock-period uint)
+        Cl.uint(this.period),
+        // (signer-sig (optional (buff 65)))
+        Cl.some(Cl.bufferFromHex(signerSig)),
+        // (signer-key (buff 33))
+        Cl.bufferFromHex(this.wallet.signerPubKey),
+        // (max-amount uint)
+        Cl.uint(maxAmount),
+        // (auth-id uint)
+        Cl.uint(this.authId),
+      ],
+      this.wallet.stxAddress,
+    );
+
+    const { result: rewardCycle } = real.network.callReadOnlyFn(
+      "ST000000000000000000002AMW42H.pox-4",
+      "burn-height-to-reward-cycle",
+      [Cl.uint(real.network.blockHeight)],
+      this.wallet.stxAddress,
+    );
+    assert(isClarityType(rewardCycle, ClarityType.UInt));
+
+    const { result: unlockBurnHeight } = real.network.callReadOnlyFn(
+      "ST000000000000000000002AMW42H.pox-4",
+      "reward-cycle-to-burn-height",
+      [Cl.uint(Number(rewardCycle.value) + this.period + 1)],
+      this.wallet.stxAddress,
+    );
+    assert(isClarityType(unlockBurnHeight, ClarityType.UInt));
+
+    // Assert
+    expect(stackStx.result).toBeOk(
+      Cl.tuple({
+        "lock-amount": Cl.uint(amountUstx),
+        "signer-key": Cl.bufferFromHex(this.wallet.signerPubKey),
+        "stacker": Cl.principal(this.wallet.stxAddress),
+        "unlock-burn-height": Cl.uint(Number(unlockBurnHeight.value)),
+      }),
+    );
+
+    // Update model so that we know this wallet is stacking. This is important
+    // in order to prevent the test from stacking multiple times with the same
+    // address.
+    model.stackers.add(this.wallet.stxAddress);
+
+    // TODO (moodmosaic): Update the wallet's balance in the model.
+    // model.wallets.set(this.wallet.stxAddress, model.wallets.get(this.wallet.stxAddress)! - amountUstx);
```

Comment:
This is the very next step once we address https://github.com/stacks-network/stacks-core/pull/4550/files#r1527141022 so that the tests can run in this branch.